src/components/Capture/index.js

```diff
@@ -106,54 +109,56 @@ export default class Capture extends Component {
     this.createCapture({...payload, valid: true})
   }
 
-  renderCapture = (useCapture) => {
-    const actions = {
-      handleMessages: this.handleMessages,
-      handleImage: this.handleImage,
-      setUploadState: this.setUploadState
-    }
-
-    let cameraProps = {onUserMedia: () => this.onUserMedia(), ...this.props};
-    const captureComponent = useCapture ?
-      <Camera {...cameraProps} {...actions} {...this.state} /> :
-      <Uploader {...this.props} {...actions} {...this.state} />
+  onUploadFallback = (file) => {
+    this.setState({uploadFallback: true})
+    this.deleteCaptures()
+    this.onImageFileSelected(file)
+  }
 
-    return (
-      <div>
-        {functionalSwitch(this.props.method, {
-          document: () => <DocumentTitle useCapture={useCapture} />,
-          face: ()=> <FaceTitle useCapture={useCapture} />
-        })}
-        {captureComponent}
-      </div>
-    )
+  onImageFileSelected = (file) => {
+    fileToBase64(file, this.handleImage)
   }
 
-  render () {
-    const {
-      method,
-      documentCaptured,
-      faceCaptured,
-      useWebcam
-    } = this.props
+  deleteCaptures = ()=> {
+    const {method, actions: {deleteCaptures}} = this.props
+    deleteCaptures({method})
+  }
 
-    const useCapture = (useWebcam && this.supportsWebcam() && isDesktop)
-    const hasCaptured = {
-      'document': documentCaptured,
-      'face': faceCaptured
-    }
-    const classes = classNames({
-      [style.camera]: useCapture && !hasCaptured[method],
-      [style.uploader]: !useCapture && !hasCaptured[method]
-    })
+  render ({method, isThereAValidCapture, useWebcam, hasUnprocessedCaptures, areAllCapturesInvalid, ...other}) {
+    const useCapture = (!this.state.uploadFallback && useWebcam && this.supportsWebcam() && isDesktop)
+    const hasCaptured = isThereAValidCapture[method]
+    //console.log(other)
     return (
-      <div className={classes}>
-        {hasCaptured[method] && <Confirm {...this.props} /> || this.renderCapture(useCapture)}
-      </div>
+      <CaptureScreen {...{method, useCapture, hasCaptured,
+        onUserMedia: this.onUserMedia, onScreenshot: this.handleImage,
+        onUploadFallback: this.onUploadFallback, onImageSelected: this.onImageFileSelected,
+        uploading:hasUnprocessedCaptures[method], noDocument: areAllCapturesInvalid[method],
+        ...other}}/>
     )
   }
 }
 
+const CaptureScreen = ({method, useCapture, hasCaptured, ...other})=> (
+  <div className={classNames({
+    [style.camera]: useCapture && !hasCaptured,
+    [style.uploader]: !useCapture && !hasCaptured
+  })}>
+  {hasCaptured ?
```

Comment:
I find this ternary syntax confusing when used like this, quite hard to read

It goes against [the style guide](https://github.com/airbnb/javascript#comparison--nested-ternaries) too, really

I'd rather the logic was moved out somehow